aws/resource_aws_lex_intent.go

```diff
@@ -0,0 +1,319 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lexmodelbuildingservice"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsLexIntent() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLexIntentCreate,
+		Read:   resourceAwsLexIntentRead,
+		Update: resourceAwsLexIntentUpdate,
+		Delete: resourceAwsLexIntentDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"checksum": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"conclusion_statement": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     lexStatementResource,
+			},
+			"confirmation_prompt": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     lexPromptResource,
+			},
+			"created_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      lexDescriptionDefault,
+				ValidateFunc: validation.StringLenBetween(lexDescriptionMinLength, lexDescriptionMaxLength),
+			},
+			"dialog_code_hook": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     lexCodeHookResource,
+			},
+			"follow_up_prompt": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     lexFollowUpPromptResource,
+			},
+			// optional for create but required on update
+			"fulfillment_activity": {
+				Type:     schema.TypeList,
+				Required: true,
```

Comment:
If I try to create an intent without a `fulfillment_activity` I get `Error: aws_lex_intent.order_flowers: "fulfillment_activity": required field is not set`, please remove comment saying it's optional.

Reply:
I was trying to indicate that it is set to required because updates require it even though the spec in Amazon API docs say it is optional. So if someone is  updating this code they don't think I did that by mistake. Would this be better? Or still think I should remove?

"This field is set to required because it is required by updates even though it is optional for creates"

Reply:
How about `Must be required because required by updates even though optional for creates`?

Reply:
Works for me.